core/store/src/trie/mod.rs

```diff
@@ -664,6 +695,17 @@ impl Trie {
                 Err(_) => " ".repeat(key.len()),
             };
             let state_record = StateRecord::from_raw_key_value(key.clone(), value);
+            let record_type_string = state_record.as_ref().expect("").get_type_string();
+
+            if let Some(s) = type_str {
+                if s != record_type_string {
+                    continue;
+                }
+            }
+
+            if let Some(ref mut val) = limit {
```

Comment:
nit: rename val to limit, it's common practice to keep the same name when unpacking option

Reply:
oh actually if you change limit to u32 you will no longer need to unpack it